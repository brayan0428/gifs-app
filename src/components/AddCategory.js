import React, { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ addCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputValue = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length === 0) {
      return;
    }
    addCategory(inputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputValue} />
    </form>
  );
};

AddCategory.propTypes = {
  addCategory: PropTypes.func.isRequired,
};
