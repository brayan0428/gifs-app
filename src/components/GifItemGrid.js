import React from "react";

export const GrifItemGrid = ({ img, title }) => (
  <div className="GifItemCard">
    <img className="GifItemCard__img" src={img} alt={title} />
    <div className="GifItemCard__body">
      <h4>{title}</h4>
    </div>
  </div>
);
