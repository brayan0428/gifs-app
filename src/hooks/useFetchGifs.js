import { useState, useEffect } from "react";
import { getGifs } from "../services";

export default (category) => {
  const [state, setState] = useState({
    gifs: [],
    loading: true,
  });
  useEffect(() => {
    getGifs(category).then((gifs) => {
      setState({ gifs, loading: false });
    });
  }, [category]);
  return state;
};
