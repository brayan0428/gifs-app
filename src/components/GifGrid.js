import React from "react";
import { GifItemGrid } from "./GifItemGrid";
import useFetchGifs from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { gifs, loading } = useFetchGifs(category);
  return (
    <>
      <h2>{category}</h2>
      {loading && <h4>Loading...</h4>}
      <div className="GifGrid">
        {gifs.map((gif) => (
          <GifItemGrid key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
