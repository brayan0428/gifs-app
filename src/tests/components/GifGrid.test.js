import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("<GifGrid />", () => {
  const category = "Goku";

  test("Validando el componente", () => {
    useFetchGifs.mockReturnValue({
      gifs: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Validando el componente con data cargada desde el useFetchGifs", () => {
    const gifs = [
      {
        id: "123",
        title: "Prueba",
        img: "http://prueba.com",
      },
    ];
    useFetchGifs.mockReturnValue({
      gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find("h4").exists()).toBe(false);
    expect(wrapper.find("GifItemGrid").length).toBe(gifs.length);
  });
});
