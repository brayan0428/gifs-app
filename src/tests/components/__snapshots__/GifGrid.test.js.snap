// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> Validando el componente 1`] = `
<Fragment>
  <h2>
    Goku
  </h2>
  <h4>
    Loading...
  </h4>
  <div
    className="GifGrid"
  />
</Fragment>
`;
